### PR TITLE
add way to set default ordering for child pages in wagtail

### PR DESCRIPTION
**Pull request by smark-1**
Add a way to set the default sort options for all child pages in the wagtail page view.

for example, in the bakerydemo BreadsIndexPage I can now add:

```
class BreadsIndexPage(Page):
   ...
   # defaults to -latest_revision_created_at
   default_order = ''ord" # can be all the valid orderings ('title', 'live', '-live', '-title', ...)
   
   # or can set method get_default_order() for more control
   def get_default_order(self):
      """"
      orders the children of the page by ord (allows reordering the page) if less then 20 child pages
      if 20 or more pages use the default setting
      """
       if Page.objects.child_of(self).count() < 20:
           return 'ord'
       return '-latest_revision_created_at'
 
```

It is possible that there is a way to do it in wagtail already, but I was not able to find it looking through the docs.

_Please check the following:_

-   [x] Do the tests still pass?[^1]
-   [x] Does the code comply with the style guide?
    -   [x] Run `make lint` from the Wagtail root.
-   [x] For Python changes: Have you added tests to cover the new/fixed behaviour?
-   [x] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    -   [x] **Please list the exact browser and operating system versions you tested**:
    -   [x] **Please list which assistive technologies [^3] you tested**:
-   [x] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**.

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

**Comment by smark-1** [comment]
@gasman
I added tests and docs? What else needs to be changed?

File: wagtail/models/__init__.py [hunk 1]
```diff
@@ -1362,6 +1362,17 @@ def get_default_locale(self):
 
         return super().get_default_locale()
 
+    # set the default order for child pages to be sorted in wagtail
+    default_order = "-latest_revision_created_at"
```
Comment (lb-) [inline wagtail/models/__init__.py:1366]: Just a small one - could we move this up alongside other class properties?

Maybe just after

```
max_count_per_parent = None
```

**Comment by lb-** [comment]
@smark-1 this is looking solid now & I have marked as needs review.

Thanks for adding tests & docs.

File: wagtail/admin/views/pages/listing.py [hunk 2]
```diff
@@ -321,6 +321,19 @@ def get_context_data(self, **kwargs):
         )
         return context
 
+    def get_ordering(self):
+        """
+        if page model has a get_default_order() method or default_order property
+        use it for the default ordering of the child pages
```
Comment (lb-) [inline wagtail/admin/views/pages/listing.py:327]:
```suggestion
        Use the parent Page's `get_admin_default_ordering` method.
```

Let's keep this short and sweet, no need to re-write what the method is doing, we could even avoid this docstring probably but happy to have it here.

**Review by lb- — CHANGES_REQUESTED** [review]
This looks good and works well.

I recommend we rename this whole thing to `get_admin_default_ordering` and `admin_default_ordering` for the attribute.

This aligns with the `get_admin_display_title` approach and helps us differentiate between what's shown in the admin and other queries that may be made on pages.

Additionally, the Page listing index (and Django index views) use the full name `ordering` not `order` so let's use that.

If you could do a rework of this PR with the suggested changes (feel free to tweak) and then the renaming of the methods/attributes (including in docs) I think this would be pretty much good to go.

**Review by lb- — APPROVED** [review]
This is looking and working great, thank you @smark-1 

I have one more docs suggestion but will merge this when I rebase.

I will force push to your branch with the commits squashed and with release notes but will wait until a few more UK time-zone team come online in case they have any strong opinions on the naming.

We should be pretty good to go though. We really appreciate you taking the time to document and write tests for this. A pretty awesome first contribution.

I will add you to the contributors list as `Shlomo Markowitz` but please advise if you want a different name listed.

**Comment by lb-** [comment]
Thank you @smark-1 - this is a really great feature.

**Pull request merged**